### DISCUSSION TITLE
What's the future of B&W radio in EdgeTx?

### DISCUSSION THREAD

**Discussion by MicheleVilla** [EdgeTX/edgetx#3096] [General]
the developers are fully concentrated in the development of radiomaster T16 and more generally in radios with color display
And I stress they are doing a great job
but for older radios with black and white display is there still leeway?
My X9E has issues with edgetx and I'm wondering if it shouldn't be frozen at OpenTx 2.3.15
In any case, thank you for the immense work you are doing

**Comment by gagarinlg** [comment]
I actually tried to debug the switch warning problem last weekend, but it does not happen in the simulator and I do not have a real X9E to test.
Currently the state is, that the radios are still supported and there is no plan to stop support for B&W radios. We will not accept NEW radios with STM32F2 MCUs and there will come a point in time when there will be no more new features for the radios with 512kB of internal flash memory. When this happens is not clear, as of now.

**Comment by pfeerick** [comment]
Quite good actually. There have been incremental improvements for B&W, and features still being added. Plus there are new radios coming out still with B&W/single color displays. While the focus has been more on colour (and touch) radios since they're the cool toy on the block, there has also been recent work for the B&W radios to reduce the size of the firmware, and reduce SRAM usage, in order to further extend the life of them. Plus there are plans to revamp at least certain parts of the B&W UI. Funnily, enough, like Malte, I was looking at the X9E issue a few days ago but then got distracted with other things (again). And similarly, I don't have an X9E to do any hardware testing or debugging. So it's not being ignored, just in the "awkward problem to diagnose" stage atm.

**Comment by MicheleVilla** [comment]
Thank you so much for the extremely comprehensive replies
I make myself available right now to test the firmware changes on my radios if it can be useful.
Of course when you have the time to do them...
I have also recently owned a T16 radiomaster and am enthusiastically following its development

**Reply by gagarinlg** [reply]
The issue with this issue is, that you need a debugger connected to the radio to test stuff.
